instrumentation/dropwizard/dropwizard-testing/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizard/DropwizardTest.java

```diff
@@ -0,0 +1,224 @@
+import io.dropwizard.setup.Environment;
+import io.dropwizard.testing.ConfigOverride;
+import io.dropwizard.testing.DropwizardTestSupport;
+import io.opentelemetry.instrumentation.api.internal.HttpConstants;
+import io.opentelemetry.instrumentation.test.utils.PortUtils;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpServerTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerTestOptions;
+import io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Locale;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class DropwizardTest extends AbstractHttpServerTest<DropwizardTestSupport<Configuration>> {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpServerInstrumentationExtension.forAgent();
+
+  @Override
+  protected DropwizardTestSupport<Configuration> setupServer() {
+    DropwizardTestSupport<Configuration> testSupport =
+        new DropwizardTestSupport<>(
+            testApp(),
+            null,
+            ConfigOverride.config("server.applicationConnectors[0].port", String.valueOf(port)),
+            ConfigOverride.config(
+                "server.adminConnectors[0].port", String.valueOf(PortUtils.findOpenPort())));
+    testSupport.before();
+    return testSupport;
+  }
+
+  @Override
+  protected void stopServer(DropwizardTestSupport<Configuration> server) {
+    server.after();
+  }
+
+  protected void customizeTestOptions(HttpServerTestOptions options) {
+    // this override is needed because dropwizard reports peer ip as the client ip
+    options.setSockPeerAddr(serverEndpoint -> TEST_CLIENT_IP);
+    options.setHasHandlerSpan(endpoint -> endpoint != NOT_FOUND);
+    options.setHasResponseSpan(endpoint -> endpoint == NOT_FOUND);
+    options.setTestPathParam(true);
+    options.setResponseCodeOnNonStandardHttpMethod(405);
+
+    options.setExpectedHttpRoute(
+        (endpoint, method) -> {
+          if (HttpConstants._OTHER.equals(method)) {
+            return getContextPath() + "/*";
+          }
+
+          if (NOT_FOUND.equals(endpoint)) {
+            return getContextPath() + "/*";
+          } else if (PATH_PARAM.equals(endpoint)) {
+            return getContextPath() + "/path/{id}/param";
+          } else {
+            return super.expectedHttpRoute(endpoint, method);
+          }
+        });
+  }
```

Comment:
We usually just place this inside the `configure` method. For the async test you currently run this block twice, once from calling `super.configure(options)` and also by explicitly calling `customizeTestOptions`.